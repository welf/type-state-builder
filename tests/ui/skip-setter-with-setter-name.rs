use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct InvalidStruct {
    name: String,
    
    // This should be an error: can't have custom setter name on skipped setters
    #[builder(default = "42", skip_setter, setter_name = "custom_name")]
    count: u32,
}

fn main() {
    let _instance = InvalidStruct::builder()
        .name("test".to_string())
        .build();
}