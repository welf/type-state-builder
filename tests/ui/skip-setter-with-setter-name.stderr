error: Field 'count' has conflicting attributes: skip_setter and setter_name
       note: #[builder(skip_setter)] and #[builder(setter_name)] are incompatible
       help: remove one of incompatible attributes
 --> tests/ui/skip-setter-with-setter-name.rs:9:5
  |
9 |     count: u32,
  |     ^^^^^
